src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -217,18 +231,55 @@ private int LookupAggregatorStore(string[] tagKeys, object[] tagValues, int leng
             return aggregatorIndex;
         }
 
+        private bool TryGetUnusedMetricPoint(out int index)
+        {
+            if (this.metricPointIndex < MaxMetricPoints)
+            {
+                index = Interlocked.Increment(ref this.metricPointIndex);
```

Comment:
Could you add a comment like to explain why we still need Interlock here.
"Interlocked.Increment required as the lock is only taken on the value2metrics, so another thread with different tag keys could be running in parallel."

Reply:
@utpilla FYI - Now i recollect why we still needed Interlocked.